openedx_tagging/core/tagging/import_export/exceptions.py

```diff
@@ -63,8 +72,8 @@ class InvalidFormat(TagParserError):
     Exception used when there is an error with the format
     """
 
-    def __init__(self, tag: dict, format: str, message: str, **kargs):
-        self.tag = tag
+    def __init__(self, tag: dict | None, format: str, message: str, **kargs):
+        super().__init__(tag)
```

Comment:
These exception classes are quite messy and could use a refactor. Once you add type hints, the problems become more obvious. In particular, none of them were calling `super().__init__()`, and the purpose of the `tag` field is unclear. Some have hard-coded messages and some let you override the `message`. For now I just made the minimal changes to get type checks passing.